pkg/resolution/resolver/bundle/bundle.go

```diff
@@ -194,7 +194,7 @@ func readTarLayer(layer v1.Layer) ([]byte, error) {
 	}
 
 	contents := make([]byte, header.Size)
```

Comment:
Nope it has the right size.

Reply:
Then why does it not read all the data into the buffer of the right size?

Reply:
> Read reads up to len(p) bytes into p. It returns the number of bytes
read (0 <= n <= len(p)) and any error encountered. ... If some data is
available but not len(p) bytes, Read conventionally returns what is
available instead of waiting for more.

`io.Reader` doesn't give you that guarantee. Why ? I am not sure 🙃